app/src/main/java/app/insti/adapter/ComplaintsRecyclerViewAdapter.java

```diff
@@ -0,0 +1,170 @@
+                    bundle.putString("sessionId", sessionID);
+                    bundle.putString("userId", userID);
+                    ComplaintDetailsFragment complaintDetailsFragment = new ComplaintDetailsFragment();
+                    complaintDetailsFragment.setArguments(bundle);
+                    AppCompatActivity activity = (AppCompatActivity) context;
+                    activity.getSupportFragmentManager().beginTransaction().replace(R.id.framelayout_for_fragment, complaintDetailsFragment).addToBackStack(TAG).commit();
+                }
+            });
+
+            Venter.Complaint complaint = complaintList.get(position);
+            try {
+                textViewDescription.setText(complaint.getDescription());
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+            try {
+                textViewLocation.setText(complaint.getLocationDescription());
+
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+            try {
+                textViewUserName.setText(complaint.getComplaintCreatedBy().getUserName());
+
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+            try {
+                textViewStatus.setText(complaint.getStatus().toUpperCase());
+                if (complaint.getStatus().equalsIgnoreCase("Reported")) {
+                    textViewStatus.setBackgroundColor(Color.parseColor("#FF0000"));
+                    textViewStatus.setTextColor(context.getResources().getColor(R.color.primaryTextColor));
+                } else if (complaint.getStatus().equalsIgnoreCase("In Progress")) {
+                    textViewStatus.setBackgroundColor(context.getResources().getColor(R.color.colorSecondary));
+                    textViewStatus.setTextColor(context.getResources().getColor(R.color.secondaryTextColor));
+                } else if (complaint.getStatus().equalsIgnoreCase("Resolved")) {
+                    textViewStatus.setBackgroundColor(Color.parseColor("#00FF00"));
+                    textViewStatus.setTextColor(context.getResources().getColor(R.color.secondaryTextColor));
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+            try {
+                String time = DateTimeUtil.getDate(complaint.getComplaintReportDate().toString());
+                Log.i(TAG, "time: " + time);
+                textViewReportDate.setText(time);
+
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+            try {
+                buttonComments.setText("COMMENTS(" + complaint.getComment().size() + ")");
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+            try {
+                buttonVotes.setText("UP VOTES(" + complaint.getUsersUpVoted().size() + ")");
+            } catch (Exception e) {
+                e.printStackTrace();
```

Comment:
Put everything in one try - catch block instead of this and do null checks explicitly.

Reply:
Will do